WordPress/src/main/java/org/wordpress/android/ui/plans/PlansActivity.java

```diff
@@ -301,14 +349,50 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
-    private void startPurchaseProcess() {
-        // TODO: this should start the Google Play purchase process, for now it shows the
-        // post-purchase on-boarding
-        boolean isBusinessPlan = (mViewPager.getCurrentItem() == mViewPager.getAdapter().getCount() - 1);
-        Intent intent = new Intent(this, PlanPostPurchaseActivity.class);
-        intent.putExtra(PlanPostPurchaseActivity.ARG_IS_BUSINESS_PLAN, isBusinessPlan);
-        startActivity(intent);
-        finish();
+    private void startPurchaseProcess(int position) {
+        final Plan plan = getPageAdapter().getPlan(position);
+        String sku = plan.getAndroidSKU();
+        Blog currentBlog = WordPress.getBlog(mLocalBlogID);
+        JSONObject extraData = new JSONObject();
+        try {
+            extraData.put("blog_id", currentBlog.getDotComBlogId());
+            extraData.put("user_id", AccountHelper.getDefaultAccount().getUserId());
+        } catch (JSONException e) {
+            AppLog.e(AppLog.T.PLANS, "Can't add extra info to purchase data!", e);
+            return;
+        }
+
+        mIabHelper.launchSubscriptionPurchaseFlow(this, sku, PURCHASE_PLAN_REQUEST,
+                new IabHelper.OnIabPurchaseFinishedListener() {
+                    public void onIabPurchaseFinished(IabResult result, Purchase info) {
+                        if (result != null) {
+                            AppLog.d(AppLog.T.PLANS, "IabResult: " + result.toString());
+                            if (result.isSuccess()) {
+                                if (info != null) {
+                                    AppPrefs.setInAppPurchaseRefreshRequired(true);
```

Comment:
We want to synch the purchase with the wpcom backend as soon as it's made on the Google Store. So the product is enabled on the site immediately after.

That app setting, and the call that happens when the app is brought in foreground, are there for security reasons, if something bad happens during the post purchase synch we will re-synch it later. Without this initial synch the backend doesn't have any info about the purchase.